From: Jeff Garzik <jgarzik@redhat.com>
Subject: [RHEL5 patch] SATA AHCI: support AHCI class code
Date: Mon, 15 Jan 2007 11:57:20 -0500
Bugzilla: 222674
Message-Id: <20070115165720.GV31287@devserv.devel.redhat.com>
Changelog: SATA AHCI: support AHCI class code


This last-minute patch helps future-proof the AHCI driver, by supporting
the AHCI PCI class code in addition to specific PCI IDs.


---
 drivers/scsi/ahci.c |    4 ++++
 1 file changed, 4 insertions(+)

Index: linux-2.6.18/drivers/scsi/ahci.c
===================================================================
--- linux-2.6.18.orig/drivers/scsi/ahci.c
+++ linux-2.6.18/drivers/scsi/ahci.c
@@ -350,6 +350,10 @@ static const struct pci_device_id ahci_p
 	{ PCI_VENDOR_ID_NVIDIA, 0x044f, PCI_ANY_ID, PCI_ANY_ID, 0, 0,
 	  board_ahci },		/* MCP65 */
 
+	/* Generic, PCI class code for AHCI */
+	{ PCI_ANY_ID, PCI_ANY_ID, PCI_ANY_ID, PCI_ANY_ID,
+	  0x010601, 0xffffff, board_ahci },
+
 	{ }	/* terminate list */
 };
 
